mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -566,8 +582,8 @@ class Word2VecModel private[spark] (
     wordList.zip(cosVec)
       .toSeq
       .sortBy(-_._2)
-      .take(num + 1)
-      .tail
+      .filter(tup => wordOpt.map(w => !w.equals(tup._1)).getOrElse(true) && tup._2 != 1.0d)
```

Comment:
Yes, those are both valid points.  I went back and forth on this one, but I think we could actually argue that rejecting the identical vector doesn't make sense in any case.